libnmstate/validator.py

```diff
@@ -251,9 +251,23 @@ def _assert_vxlan_has_missing_attribute(state, *attributes):
 
 def _assert_vlan_filtering_trunk_tags(ports_state):
     for port_state in ports_state:
-        trunk_tags = port_state.get(LB.Port.VLAN_SUBTREE, {}).get(
-            LB.Port.Vlan.TRUNK_TAGS, []
-        )
+        port_vlan_state = port_state.get(LB.Port.VLAN_SUBTREE, {})
+        port_type = port_vlan_state[LB.Port.Vlan.TYPE]
+        trunk_tags = port_vlan_state.get(LB.Port.Vlan.TRUNK_TAGS, [])
+
+        if port_type == LB.Port.Vlan.ACCESS_TYPE:
+            if trunk_tags:
+                raise NmstateValueError('Access port cannot have trunk tags')
+        else:
+            if len(trunk_tags) == 1:
+                if not trunk_tags[0].get(LB.Port.Vlan.TrunkTags.ID_RANGE):
```

Comment:
Is it ok if l `len(trunk_tags` is 2 and `trunk_tags[0].get(LB.Port.Vlan.TrunkTags.ID_RANGE)` and `trunk_tags[1].get(LB.Port.Vlan.TrunkTags.ID_RANGE)` are both False or does this need to check all items of `trunk_tags`?

Reply:
The important thing here is that we have at least 2 trunk tags in the port vlan config.

Hence, 1 tag range is OK; >= 2 single tags are also OK. Unless I'm mistaken, the latter is the scenario you're asking about.

Reply:
Can you adjust the error message to say something like `A trunk port needs to specify a vlan tag range or multiple tags` to clarify this?

Reply:
Sure. Done.